### ISSUE TITLE
Cannot specify a single usage to key

### ISSUE THREAD

**Issue by PhoenixBlue** [OpenSC/OpenSC#1651]
### Problem Description
Cannot generate key pair with a single usage (encrypt/decrypt only, for example)

### Steps to reproduce
pkcs11-tool --module /usr/lib/x86_64-linux-gnu/softhsm/libsofthsm2.so -p 12345678  -k --key-type rsa:1024 --id aabbcc --usage-sign

### Result
Private Key Object; RSA 
  label:      
  ID:         aabbcc
  Usage:      decrypt, sign, unwrap
Public Key Object; RSA 1024 bits
  label:      
  ID:         aabbcc
  Usage:      encrypt, verify, wrap

### Expected result

Private Key Object; RSA 
  label:      
  ID:         aabbcc
  Usage:       sign
Public Key Object; RSA 1024 bits
  label:      
  ID:         aabbcc
  Usage:       verify

**Comment by frankmorgner** [comment]
That behavior is expected and documented in [the manual](https://htmlpreview.github.io/?https://github.com/OpenSC/OpenSC/blob/master/doc/tools/tools.html#pkcs11-tool). Why do you want to have `--usage-verify` (which doesn't currently exist)?

**Comment by PhoenixBlue** [comment]
> 
> 
>  Why do you want to have `--usage-verify` (which doesn't currently exist)?

My bad! I meant `--usage-sign`. I modified the post. Thanks for the answer. 
Actually, I tried to write my own code that would sign some data and perform verification of such signature.  I am using the `softhsm/pkcs11`  library. I always ended up with error: `Error at Encrypt init: 5`.
So from hours of trying to search for an answer, I read that the certificate must be set for signing only. I am not sure why. I wanted to gice it a try.
